services/src/main/scala/cromwell/services/instrumentation/CromwellInstrumentation.scala

```diff
@@ -21,19 +20,73 @@ object CromwellInstrumentation {
     .as[Option[FiniteDuration]]("instrumentation-rate")
     .getOrElse(5.seconds)
 
-  type InstrumentationPath = NonEmptyList[String]
-
-  implicit def stringToNel(str: String): NonEmptyList[String] = NonEmptyList.of(str)
-
-  implicit class EnhancedStatsDPath(val path: InstrumentationPath) extends AnyVal {
-    def withStatusCodeFailure(code: Option[Int]) = code
-      .map(c => path.concatNel(NonEmptyList.of(c.toString)))
-      .getOrElse(path)
+  /**
+   * A representation of a metric name, with different output formats for different kinds of instrumentation services.
+   * Those designed to handle one time series per metric name can get a simple list of name parts with [[getFlatPath]],
+   * while those designed to handle multiple time series per metric name can get a shorter metric name and
+   * frequently-changing, high-variant "labels" separately via [[getPathAndLabels]].
+   *
+   * This distinction exists because instrumentation services meant to handle multiple time series per metric name
+   * (like Prometheus) are poorly built to store numerous metric names or query across them. The more complex
+   * [[getPathAndLabels]] output allows the backends for those instrumentation services to "play nice" with their
+   * associated ecosystems.
+   *
+   * @param internalPath the runtime representation of this class, where the 'left' is a simple, always-present
+   *                     low-variant name part, and the 'right' is a key-value pair for a frequently-changing
+   *                     high-variant name part
+   */
+  implicit class InstrumentationPath (val internalPath: NonEmptyList[Either[String, (String, String)]]) extends AnyVal {
```

Comment:
Makes sense, I added some internal types to that class and used those for documentation. I figured I wouldn't expose them outside the class since I think using them as method inputs or something would just be unnecessary indirection